src/compiler/moduleNameResolver.ts

```diff
@@ -362,9 +362,6 @@ namespace ts {
         }
 
         function getOrCreateCacheForModuleName(nonRelativeModuleName: string): PerModuleNameCache {
-            if (isExternalModuleNameRelative(nonRelativeModuleName)) {

```

Comment:
Instead of removing I suggest you put the assert here so that we can catch incorrect use of the cache in future.